packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -18,6 +20,53 @@ moduleFor('Components test: curly components', class extends RenderingTest {
     this.assertComponentElement(this.firstChild, { content: 'hello' });
   }
 
+  ['@htmlbars it can have a custom id and it is not bound']() {
+    this.registerComponent('foo-bar', { template: '{{id}} {{elementId}}' });
+
+    this.render('{{foo-bar id=customId}}', {
+      customId: 'bizz'
+    });
+
+    this.assertComponentElement(this.firstChild, { tagName: 'div', attrs: { id: 'bizz' }, content: 'bizz bizz' });
+
+    this.runTask(() => this.rerender());
+
+    this.assertComponentElement(this.firstChild, { tagName: 'div', attrs: { id: 'bizz' }, content: 'bizz bizz' });
+
+    this.runTask(() => set(this.context, 'customId', 'bar'));
+
+    this.assertComponentElement(this.firstChild, { tagName: 'div', attrs: { id: 'bizz' }, content: 'bar bizz' });
+
+    this.runTask(() => set(this.context, 'customId', 'bizz'));
+
+    this.assertComponentElement(this.firstChild, { tagName: 'div', attrs: { id: 'bizz' }, content: 'bizz bizz' });
+  }
+
+  ['@htmlbars elementId cannot change'](assert) {
+    let component;
+    let FooBarComponent = Component.extend({
+      elementId: 'blahzorz',
+      init() {
+        this._super(...arguments);
+        component = this;
+      }
+    });
+
+    this.registerComponent('foo-bar', { ComponentClass: FooBarComponent, template: '{{elementId}}' });
+
+    this.render('{{foo-bar}}');
+
+    this.assertComponentElement(this.firstChild, { tagName: 'div', attrs: { id: 'blahzorz' }, content: 'blahzorz' });
+
+    if (EmberDev && !EmberDev.runningProdBuild) {
```

Comment:
If only https://github.com/emberjs/ember.js/blob/master/packages/ember-views/lib/views/states/in_dom.js#L24 used `assert` this wouldn't be needed (you could use `expectAssertion` which does these checks internally).

This is good though, I'm just grumbling :smiling_imp: